Change charms to use juju-http{s}-proxy model settings

VariableDeclared opened the pull request:
Changed charm to use juju-http{s}-proxy-settings when these keys are defined.

[LP#1831712](https://bugs.launchpad.net/charm-kubernetes-worker/+bug/1831712)

Commit:
Refactored method to container-runtime-common

Moved method from container runtime charms to
this layer, also added logic to parse CIDR
subnets.

Commit:
Refactored and added unit tests for CIDR notation

Commit:
Remove extraneous files and trim call

Commit:
Add lower case key

Added lower case no_proxy key to conver cases
where template files look for this lower case
version.

Commit:
Remove print statement.

Commit:
Refactored and added check for None

Renamed a variable and added a check for
the case where juju environement variables are None.

Commit:
Added test for empty and docs

Added test for empty return and added tests
section to README.

Commit:
Changed precedent on environment config

Added tests and changed logic so that
local charm config has precedent over environment.

Commit:
Discovered an issue and created a test for it

Commit:
Added key to disable juju-https-config

Added config key for case where a user
might want to disable juju-proxy for this specific charm but does
not want to set another proxy.

Commit:
Fix error with config key

changed disable-proxy for disable-juju-proxy.

Cynerva commented on lib/charms/layer/container_runtime_common.py:
```diff
@@ -13,6 +17,61 @@
+
+def check_for_juju_https_proxy(config):
+    # If juju environment variables are defined, take precedent
+    # over config.yaml.
```
This comment is no longer correct, right?

VariableDeclared replied:
yup, cheers for spotting that!

Cynerva commented on lib/charms/layer/container_runtime_common.py:
```diff
@@ -13,6 +17,61 @@
+            except ValueError:
+                hosts.append(address)
+        parsed_hosts = ",".join(hosts)
+        return parsed_hosts
```
I like how you did this :+1:

VariableDeclared replied:
:smile:

Commit:
Review changes

Changed precedent comment

Commit:
Fix merging of config

Discovered an issue where merging did not unify config keys

Fixed this and updated tests.

Cynerva commented:
LGTM :+1:
